In [1]:
import pandas as pd
import numpy as np

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_columns', None)

mpl.rcParams['axes.unicode_minus'] = False
plt.rcParams['font.family'] = 'Malgun Gothic'


pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

In [2]:
df = pd.read_csv('./1년 1시간별 Ls, Ws.csv', encoding = 'cp949')
df

자치구           대여소명  대여소번호     년  월  시간    lambda        mu    c  \
0       중랑구  (구)신한은행 중랑교지점   1442  2020  6   0  3.800000  2.700000  6.0   
1       중랑구  (구)신한은행 중랑교지점   1442  2020  6   1  3.700000  2.133333  6.0   
2       중랑구  (구)신한은행 중랑교지점   1442  2020  6   2  2.133333  1.266667  6.0   
3       중랑구  (구)신한은행 중랑교지점   1442  2020  6   3  1.566667  0.833333  6.0   
4       중랑구  (구)신한은행 중랑교지점   1442  2020  6   4  1.333333  0.533333  6.0   
...     ...            ...    ...   ... ..  ..       ...       ...  ...   
694939  강서구        힐스테이트에코   2720  2021  5  19  7.933333  8.066667  9.0   
694940  강서구        힐스테이트에코   2720  2021  5  20  5.166667  7.800000  9.0   
694941  강서구        힐스테이트에코   2720  2021  5  21  3.400000  6.933333  9.0   
694942  강서구        힐스테이트에코   2720  2021  5  22  1.800000  3.000000  9.0   
694943  강서구        힐스테이트에코   2720  2021  5  23  1.233333  1.166667  9.0   

             rho       p_0       p_1      p_1+       p_c       L_s        W_s  
0       1.407407  0.040995  0.057696  0.959005  0.318602  4.249820   1.641292  
1       1.734375  0.015893  0.027565  0.984107  0.432587  4.789792   2.281475  
2       1.684211  0.018275  0.030779  0.981725  0.417101  4.725431   3.800050  
3       1.880000  0.010731  0.020174  0.989269  0.473793  4.948997   6.003217  
4       2.500000  0.002462  0.006154  0.997538  0.600985  5.344821  10.046270  
...          ...       ...       ...       ...       ...       ...        ...  
694939  0.983471  0.107665  0.105886  0.892335  0.092668  4.362561   0.606066  
694940  0.662393  0.343188  0.227325  0.656812  0.008425  1.796723   0.350708  
694941  0.490385  0.510026  0.250109  0.489974  0.000836  0.954216   0.280887  
694942  0.600000  0.402433  0.241460  0.597567  0.004056  1.439166   0.802792  
694943  1.057143  0.076892  0.081286  0.923108  0.126790  4.956086   4.601924  

[694944 rows x 16 columns]

In [3]:
df = df[df['대여소명'] == '몽촌토성역 1번출구 옆']
df

자치구          대여소명  대여소번호     년   월  시간     lambda         mu     c  \
252288  송파구  몽촌토성역 1번출구 옆   1268  2020   6   0   2.766667   3.000000  15.0   
252289  송파구  몽촌토성역 1번출구 옆   1268  2020   6   1   2.166667   1.633333  15.0   
252290  송파구  몽촌토성역 1번출구 옆   1268  2020   6   2   1.400000   1.866667  15.0   
252291  송파구  몽촌토성역 1번출구 옆   1268  2020   6   3   1.100000   1.000000  15.0   
252292  송파구  몽촌토성역 1번출구 옆   1268  2020   6   4   0.800000   0.966667  15.0   
252293  송파구  몽촌토성역 1번출구 옆   1268  2020   6   5   0.433333   0.500000  15.0   
252294  송파구  몽촌토성역 1번출구 옆   1268  2020   6   6   1.200000   1.133333  15.0   
252295  송파구  몽촌토성역 1번출구 옆   1268  2020   6   7   2.900000   2.600000  15.0   
252296  송파구  몽촌토성역 1번출구 옆   1268  2020   6   8   4.700000   3.733333  15.0   
252297  송파구  몽촌토성역 1번출구 옆   1268  2020   6   9   2.533333   3.066667  15.0   
252298  송파구  몽촌토성역 1번출구 옆   1268  2020   6  10   2.366667   2.200000  15.0   
252299  송파구  몽촌토성역 1번출구 옆   1268  2020   6  11   3.733333   3.633333  15.0   
252300  송파구  몽촌토성역 1번출구 옆   1268  2020   6  12   3.466667   3.166667  15.0   
252301  송파구  몽촌토성역 1번출구 옆   1268  2020   6  13   4.366667   3.933333  15.0   
252302  송파구  몽촌토성역 1번출구 옆   1268  2020   6  14   4.700000   4.466667  15.0   
252303  송파구  몽촌토성역 1번출구 옆   1268  2020   6  15   6.133333   7.066667  15.0   
252304  송파구  몽촌토성역 1번출구 옆   1268  2020   6  16   8.033333   8.100000  15.0   
252305  송파구  몽촌토성역 1번출구 옆   1268  2020   6  17  11.000000  12.300000  15.0   
252306  송파구  몽촌토성역 1번출구 옆   1268  2020   6  18  11.300000  11.766667  15.0   
252307  송파구  몽촌토성역 1번출구 옆   1268  2020   6  19  11.733333  11.866667  15.0   
252308  송파구  몽촌토성역 1번출구 옆   1268  2020   6  20  13.500000  13.000000  15.0   
252309  송파구  몽촌토성역 1번출구 옆   1268  2020   6  21  11.933333  12.100000  15.0   
252310  송파구  몽촌토성역 1번출구 옆   1268  2020   6  22   9.433333   9.166667  15.0   
252311  송파구  몽촌토성역 1번출구 옆   1268  2020   6  23   6.033333   5.200000  15.0   
252312  송파구  몽촌토성역 1번출구 옆   1268  2020   7   0   1.933333   1.766667  15.0   
252313  송파구  몽촌토성역 1번출구 옆   1268  2020   7   1   1.700000   1.500000  15.0   
252314  송파구  몽촌토성역 1번출구 옆   1268  2020   7   2   1.233333   1.633333  15.0   
252315  송파구  몽촌토성역 1번출구 옆   1268  2020   7   3   1.066667   1.100000  15.0   
252316  송파구  몽촌토성역 1번출구 옆   1268  2020   7   4   0.500000   0.733333  15.0   
252317  송파구  몽촌토성역 1번출구 옆   1268  2020   7   5   0.433333   0.366667  15.0   
252318  송파구  몽촌토성역 1번출구 옆   1268  2020   7   6   1.400000   0.866667  15.0   
252319  송파구  몽촌토성역 1번출구 옆   1268  2020   7   7   2.533333   2.666667  15.0   
252320  송파구  몽촌토성역 1번출구 옆   1268  2020   7   8   3.866667   2.166667  15.0   
252321  송파구  몽촌토성역 1번출구 옆   1268  2020   7   9   1.966667   2.033333  15.0   
252322  송파구  몽촌토성역 1번출구 옆   1268  2020   7  10   2.066667   2.033333  15.0   
252323  송파구  몽촌토성역 1번출구 옆   1268  2020   7  11   2.166667   2.200000  15.0   
252324  송파구  몽촌토성역 1번출구 옆   1268  2020   7  12   2.366667   2.733333  15.0   
252325  송파구  몽촌토성역 1번출구 옆   1268  2020   7  13   3.833333   3.633333  15.0   
252326  송파구  몽촌토성역 1번출구 옆   1268  2020   7  14   3.633333   4.033333  15.0   
252327  송파구  몽촌토성역 1번출구 옆   1268  2020   7  15   4.200000   4.533333  15.0   
252328  송파구  몽촌토성역 1번출구 옆   1268  2020   7  16   5.366667   5.233333  15.0   
252329  송파구  몽촌토성역 1번출구 옆   1268  2020   7  17   7.333333   8.966667  15.0   
252330  송파구  몽촌토성역 1번출구 옆   1268  2020   7  18   7.800000   7.966667  15.0   
252331  송파구  몽촌토성역 1번출구 옆   1268  2020   7  19   8.833333   8.366667  15.0   
252332  송파구  몽촌토성역 1번출구 옆   1268  2020   7  20   8.100000   8.133333  15.0   
252333  송파구  몽촌토성역 1번출구 옆   1268  2020   7  21   8.266667   8.133333  15.0   
252334  송파구  몽촌토성역 1번출구 옆   1268  2020   7  22   5.100000   5.166667  15.0   
252335  송파구  몽촌토성역 1번출구 옆   1268  2020   7  23   3.433333   3.266667  15.0   
252336  송파구  몽촌토성역 1번출구 옆   1268  2020   8   0   1.500000   1.700000  15.0   
252337  송파구  몽촌토성역 1번출구 옆   1268  2020   8   1   1.466667   0.900000  15.0   
252338  송파구  몽촌토성역 1번출구 옆   12

In [4]:
df = df.drop(['대여소번호', 'c', 'rho', 'p_c', 'p_0', 'p_1'], axis = 1)
df

자치구          대여소명     년   월  시간     lambda         mu      p_1+  \
252288  송파구  몽촌토성역 1번출구 옆  2020   6   0   2.766667   3.000000  0.892904   
252289  송파구  몽촌토성역 1번출구 옆  2020   6   1   2.166667   1.633333  0.996409   
252290  송파구  몽촌토성역 1번출구 옆  2020   6   2   1.400000   1.866667  0.747469   
252291  송파구  몽촌토성역 1번출구 옆  2020   6   3   1.100000   1.000000  0.972183   
252292  송파구  몽촌토성역 1번출구 옆  2020   6   4   0.800000   0.966667  0.818813   
252293  송파구  몽촌토성역 1번출구 옆  2020   6   5   0.433333   0.500000  0.851637   
252294  송파구  몽촌토성역 1번출구 옆  2020   6   6   1.200000   1.133333  0.960669   
252295  송파구  몽촌토성역 1번출구 옆  2020   6   7   2.900000   2.600000  0.975649   
252296  송파구  몽촌토성역 1번출구 옆  2020   6   8   4.700000   3.733333  0.993329   
252297  송파구  몽촌토성역 1번출구 옆  2020   6   9   2.533333   3.066667  0.817503   
252298  송파구  몽촌토성역 1번출구 옆  2020   6  10   2.366667   2.200000  0.965826   
252299  송파구  몽촌토성역 1번출구 옆  2020   6  11   3.733333   3.633333  0.949412   
252300  송파구  몽촌토성역 1번출구 옆  2020   6  12   3.466667   3.166667  0.970900   
252301  송파구  몽촌토성역 1번출구 옆  2020   6  13   4.366667   3.933333  0.974521   
252302  송파구  몽촌토성역 1번출구 옆  2020   6  14   4.700000   4.466667  0.958493   
252303  송파구  몽촌토성역 1번출구 옆  2020   6  15   6.133333   7.066667  0.852646   
252304  송파구  몽촌토성역 1번출구 옆  2020   6  16   8.033333   8.100000  0.933552   
252305  송파구  몽촌토성역 1번출구 옆  2020   6  17  11.000000  12.300000  0.873056   
252306  송파구  몽촌토성역 1번출구 옆  2020   6  18  11.300000  11.766667  0.916793   
252307  송파구  몽촌토성역 1번출구 옆  2020   6  19  11.733333  11.866667  0.932065   
252308  송파구  몽촌토성역 1번출구 옆  2020   6  20  13.500000  13.000000  0.953613   
252309  송파구  몽촌토성역 1번출구 옆  2020   6  21  11.933333  12.100000  0.930790   
252310  송파구  몽촌토성역 1번출구 옆  2020   6  22   9.433333   9.166667  0.950032   
252311  송파구  몽촌토성역 1번출구 옆  2020   6  23   6.033333   5.200000  0.983624   
252312  송파구  몽촌토성역 1번출구 옆  2020   7   0   1.933333   1.766667  0.970801   
252313  송파구  몽촌토성역 1번출구 옆  2020   7   1   1.700000   1.500000  0.979194   
252314  송파구  몽촌토성역 1번출구 옆  2020   7   2   1.233333   1.633333  0.752335   
252315  송파구  몽촌토성역 1번출구 옆  2020   7   3   1.066667   1.100000  0.922062   
252316  송파구  몽촌토성역 1번출구 옆  2020   7   4   0.500000   0.733333  0.681123   
252317  송파구  몽촌토성역 1번출구 옆  2020   7   5   0.433333   0.366667  0.986513   
252318  송파구  몽촌토성역 1번출구 옆  2020   7   6   1.400000   0.866667  0.999714   
252319  송파구  몽촌토성역 1번출구 옆  2020   7   7   2.533333   2.666667  0.910694   
252320  송파구  몽촌토성역 1번출구 옆  2020   7   8   3.866667   2.166667  0.999926   
252321  송파구  몽촌토성역 1번출구 옆  2020   7   9   1.966667   2.033333  0.920686   
252322  송파구  몽촌토성역 1번출구 옆  2020   7  10   2.066667   2.033333  0.944831   
252323  송파구  몽촌토성역 1번출구 옆  2020   7  11   2.166667   2.200000  0.930091   
252324  송파구  몽촌토성역 1번출구 옆  2020   7  12   2.366667   2.733333  0.850982   
252325  송파구  몽촌토성역 1번출구 옆  2020   7  13   3.833333   3.633333  0.959433   
252326  송파구  몽촌토성역 1번출구 옆  2020   7  14   3.633333   4.033333  0.877859   
252327  송파구  몽촌토성역 1번출구 옆  2020   7  15   4.200000   4.533333  0.895755   
252328  송파구  몽촌토성역 1번출구 옆  2020   7  16   5.366667   5.233333  0.948594   
252329  송파구  몽촌토성역 1번출구 옆  2020   7  17   7.333333   8.966667  0.810242   
252330  송파구  몽촌토성역 1번출구 옆  2020   7  18   7.800000   7.966667  0.927107   
252331  송파구  몽촌토성역 1번출구 옆  2020   7  19   8.833333   8.366667  0.959675   
252332  송파구  몽촌토성역 1번출구 옆  2020   7  20   8.100000   8.133333  0.935557   
252333  송파구  몽촌토성역 1번출구 옆  2020   7  21   8.266667   8.133333  0.944831   
252334  송파구  몽촌토성역 1번출구 옆  2020   7  22   5.100000   5.166667  0.931229   
252335  송파구  몽촌토성역 1번출구 옆  2020   7  23   3.433333   3.266667  0.958079   
252336  송파구  몽촌토성역 1번출구 옆  2020   8   0   1.500000   1.700000  0.863995   
252337  송파구  몽촌토성역 1번출구 옆  2020   8   1   1.466667   0.900000  0.999745   
252338  송파구  몽촌토성역 1번출구 옆  2020   8   2   1.266667   1.533333  0.817503   
252339  송파구  몽촌토성역 1번출구 옆  2020   8   3   0.766667   1.100000  0.696027   
252340  송파구  몽촌토성역 1번출구 옆  2020  

In [5]:
data = pd.read_csv('./몽촌토성역 1번출구 옆 균형, 불균형.csv', encoding = 'cp949')
data

자치구          대여소명     년   월  시간        재고량  3사분위수  2사분위수     1사분위수   균형
0    송파구  몽촌토성역 1번출구 옆  2020   6   0   3.333333    8.7   5.55  3.733333  불균형
1    송파구  몽촌토성역 1번출구 옆  2020   6   1   4.033333    8.7   5.55  3.733333   균형
2    송파구  몽촌토성역 1번출구 옆  2020   6   2   3.700000    8.7   5.55  3.733333  불균형
3    송파구  몽촌토성역 1번출구 옆  2020   6   3   4.066667    8.7   5.55  3.733333   균형
4    송파구  몽촌토성역 1번출구 옆  2020   6   4   4.100000    8.7   5.55  3.733333   균형
5    송파구  몽촌토성역 1번출구 옆  2020   6   5   4.200000    8.7   5.55  3.733333   균형
6    송파구  몽촌토성역 1번출구 옆  2020   6   6   4.333333    8.7   5.55  3.733333   균형
7    송파구  몽촌토성역 1번출구 옆  2020   6   7   4.700000    8.7   5.55  3.733333   균형
8    송파구  몽촌토성역 1번출구 옆  2020   6   8   3.566667    8.7   5.55  3.733333  불균형
9    송파구  몽촌토성역 1번출구 옆  2020   6   9   3.100000    8.7   5.55  3.733333  불균형
10   송파구  몽촌토성역 1번출구 옆  2020   6  10   3.266667    8.7   5.55  3.733333  불균형
11   송파구  몽촌토성역 1번출구 옆  2020   6  11   3.366667    8.7   5.55  3.733333  불균형
12   송파구  몽촌토성역 1번출구 옆  2020   6  12   3.733333    8.7   5.55  3.733333   균형
13   송파구  몽촌토성역 1번출구 옆  2020   6  13   4.166667    8.7   5.55  3.733333   균형
14   송파구  몽촌토성역 1번출구 옆  2020   6  14   4.600000    8.7   5.55  3.733333   균형
15   송파구  몽촌토성역 1번출구 옆  2020   6  15   3.700000    8.7   5.55  3.733333  불균형
16   송파구  몽촌토성역 1번출구 옆  2020   6  16   3.733333    8.7   5.55  3.733333   균형
17   송파구  몽촌토성역 1번출구 옆  2020   6  17   2.500000    8.7   5.55  3.733333  불균형
18   송파구  몽촌토성역 1번출구 옆  2020   6  18   2.033333    8.7   5.55  3.733333  불균형
19   송파구  몽촌토성역 1번출구 옆  2020   6  19   1.900000    8.7   5.55  3.733333  불균형
20   송파구  몽촌토성역 1번출구 옆  2020   6  20   2.400000    8.7   5.55  3.733333  불균형
21   송파구  몽촌토성역 1번출구 옆  2020   6  21   2.333333    8.7   5.55  3.733333  불균형
22   송파구  몽촌토성역 1번출구 옆  2020   6  22   2.633333    8.7   5.55  3.733333  불균형
23   송파구  몽촌토성역 1번출구 옆  2020   6  23   3.500000    8.7   5.55  3.733333  불균형
24   송파구  몽촌토성역 1번출구 옆  2020   7   0   1.733333    8.7   5.55  3.733333  불균형
25   송파구  몽촌토성역 1번출구 옆  2020   7   1   1.966667    8.7   5.55  3.733333  불균형
26   송파구  몽촌토성역 1번출구 옆  2020   7   2   2.000000    8.7   5.55  3.733333  불균형
27   송파구  몽촌토성역 1번출구 옆  2020   7   3   2.266667    8.7   5.55  3.733333  불균형
28   송파구  몽촌토성역 1번출구 옆  2020   7   4   2.266667    8.7   5.55  3.733333  불균형
29   송파구  몽촌토성역 1번출구 옆  2020   7   5   2.533333    8.7   5.55  3.733333  불균형
30   송파구  몽촌토성역 1번출구 옆  2020   7   6   3.100000    8.7   5.55  3.733333  불균형
31   송파구  몽촌토성역 1번출구 옆  2020   7   7   3.066667    8.7   5.55  3.733333  불균형
32   송파구  몽촌토성역 1번출구 옆  2020   7   8   2.133333    8.7   5.55  3.733333  불균형
33   송파구  몽촌토성역 1번출구 옆  2020   7   9   2.066667    8.7   5.55  3.733333  불균형
34   송파구  몽촌토성역 1번출구 옆  2020   7  10   2.200000    8.7   5.55  3.733333  불균형
35   송파구  몽촌토성역 1번출구 옆  2020   7  11   2.300000    8.7   5.55  3.733333  불균형
36   송파구  몽촌토성역 1번출구 옆  2020   7  12   2.100000    8.7   5.55  3.733333  불균형
37   송파구  몽촌토성역 1번출구 옆  2020   7  13   2.300000    8.7   5.55  3.733333  불균형
38   송파구  몽촌토성역 1번출구 옆  2020   7  14   1.966667    8.7   5.55  3.733333  불균형
39   송파구  몽촌토성역 1번출구 옆  2020   7  15   1.833333    8.7   5.55  3.733333  불균형
40   송파구  몽촌토성역 1번출구 옆  2020   7  16   2.033333    8.7   5.55  3.733333  불균형
41   송파구  몽촌토성역 1번출구 옆  2020   7  17   0.400000    8.7   5.55  3.733333  불균형
42   송파구  몽촌토성역 1번출구 옆  2020   7  18   0.233333    8.7   5.55  3.733333  불균형
43   송파구  몽촌토성역 1번출구 옆  2020   7  19   0.700000    8.7   5.55  3.733333  불균형
44   송파구  몽촌토성역 1번출구 옆  2020   7  20   0.666667    8.7   5.55  3.733333  불균형
45   송파구  몽촌토성역 1번출구 옆  2020   7  21   0.833333    8.7   5.55  3.733333  불균형
46   송파구  몽촌토성역 1번출구 옆  2020   7  22   0.833333    8.7   5.55  3.733333  불균형
47   송파구  몽촌토성역 1번출구 옆  2020   7  23   1.200000    8.7   5.55  3.733333  불균형
48   송파구  몽촌토성역 1번출구 옆  2020   8   0   4.366667    8.7   5.55  3.733333   균형
49   송파구  몽촌토성역 1번출구 옆  2020   8   1   5.066667    8.7   5.55  3.733333   균형
50   송파구  몽촌토성역 1번출구 옆  2020   8   2   5.333333    8.7   5.55  3.733333   균형
5

In [6]:
df_new = pd.merge(df,data, on = ['자치구', '대여소명','년', '월', '시간'],how='outer')
df_new

자치구          대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  몽촌토성역 1번출구 옆  2020   6   0   2.766667   3.000000  0.892904   
1    송파구  몽촌토성역 1번출구 옆  2020   6   1   2.166667   1.633333  0.996409   
2    송파구  몽촌토성역 1번출구 옆  2020   6   2   1.400000   1.866667  0.747469   
3    송파구  몽촌토성역 1번출구 옆  2020   6   3   1.100000   1.000000  0.972183   
4    송파구  몽촌토성역 1번출구 옆  2020   6   4   0.800000   0.966667  0.818813   
5    송파구  몽촌토성역 1번출구 옆  2020   6   5   0.433333   0.500000  0.851637   
6    송파구  몽촌토성역 1번출구 옆  2020   6   6   1.200000   1.133333  0.960669   
7    송파구  몽촌토성역 1번출구 옆  2020   6   7   2.900000   2.600000  0.975649   
8    송파구  몽촌토성역 1번출구 옆  2020   6   8   4.700000   3.733333  0.993329   
9    송파구  몽촌토성역 1번출구 옆  2020   6   9   2.533333   3.066667  0.817503   
10   송파구  몽촌토성역 1번출구 옆  2020   6  10   2.366667   2.200000  0.965826   
11   송파구  몽촌토성역 1번출구 옆  2020   6  11   3.733333   3.633333  0.949412   
12   송파구  몽촌토성역 1번출구 옆  2020   6  12   3.466667   3.166667  0.970900   
13   송파구  몽촌토성역 1번출구 옆  2020   6  13   4.366667   3.933333  0.974521   
14   송파구  몽촌토성역 1번출구 옆  2020   6  14   4.700000   4.466667  0.958493   
15   송파구  몽촌토성역 1번출구 옆  2020   6  15   6.133333   7.066667  0.852646   
16   송파구  몽촌토성역 1번출구 옆  2020   6  16   8.033333   8.100000  0.933552   
17   송파구  몽촌토성역 1번출구 옆  2020   6  17  11.000000  12.300000  0.873056   
18   송파구  몽촌토성역 1번출구 옆  2020   6  18  11.300000  11.766667  0.916793   
19   송파구  몽촌토성역 1번출구 옆  2020   6  19  11.733333  11.866667  0.932065   
20   송파구  몽촌토성역 1번출구 옆  2020   6  20  13.500000  13.000000  0.953613   
21   송파구  몽촌토성역 1번출구 옆  2020   6  21  11.933333  12.100000  0.930790   
22   송파구  몽촌토성역 1번출구 옆  2020   6  22   9.433333   9.166667  0.950032   
23   송파구  몽촌토성역 1번출구 옆  2020   6  23   6.033333   5.200000  0.983624   
24   송파구  몽촌토성역 1번출구 옆  2020   7   0   1.933333   1.766667  0.970801   
25   송파구  몽촌토성역 1번출구 옆  2020   7   1   1.700000   1.500000  0.979194   
26   송파구  몽촌토성역 1번출구 옆  2020   7   2   1.233333   1.633333  0.752335   
27   송파구  몽촌토성역 1번출구 옆  2020   7   3   1.066667   1.100000  0.922062   
28   송파구  몽촌토성역 1번출구 옆  2020   7   4   0.500000   0.733333  0.681123   
29   송파구  몽촌토성역 1번출구 옆  2020   7   5   0.433333   0.366667  0.986513   
30   송파구  몽촌토성역 1번출구 옆  2020   7   6   1.400000   0.866667  0.999714   
31   송파구  몽촌토성역 1번출구 옆  2020   7   7   2.533333   2.666667  0.910694   
32   송파구  몽촌토성역 1번출구 옆  2020   7   8   3.866667   2.166667  0.999926   
33   송파구  몽촌토성역 1번출구 옆  2020   7   9   1.966667   2.033333  0.920686   
34   송파구  몽촌토성역 1번출구 옆  2020   7  10   2.066667   2.033333  0.944831   
35   송파구  몽촌토성역 1번출구 옆  2020   7  11   2.166667   2.200000  0.930091   
36   송파구  몽촌토성역 1번출구 옆  2020   7  12   2.366667   2.733333  0.850982   
37   송파구  몽촌토성역 1번출구 옆  2020   7  13   3.833333   3.633333  0.959433   
38   송파구  몽촌토성역 1번출구 옆  2020   7  14   3.633333   4.033333  0.877859   
39   송파구  몽촌토성역 1번출구 옆  2020   7  15   4.200000   4.533333  0.895755   
40   송파구  몽촌토성역 1번출구 옆  2020   7  16   5.366667   5.233333  0.948594   
41   송파구  몽촌토성역 1번출구 옆  2020   7  17   7.333333   8.966667  0.810242   
42   송파구  몽촌토성역 1번출구 옆  2020   7  18   7.800000   7.966667  0.927107   
43   송파구  몽촌토성역 1번출구 옆  2020   7  19   8.833333   8.366667  0.959675   
44   송파구  몽촌토성역 1번출구 옆  2020   7  20   8.100000   8.133333  0.935557   
45   송파구  몽촌토성역 1번출구 옆  2020   7  21   8.266667   8.133333  0.944831   
46   송파구  몽촌토성역 1번출구 옆  2020   7  22   5.100000   5.166667  0.931229   
47   송파구  몽촌토성역 1번출구 옆  2020   7  23   3.433333   3.266667  0.958079   
48   송파구  몽촌토성역 1번출구 옆  2020   8   0   1.500000   1.700000  0.863995   
49   송파구  몽촌토성역 1번출구 옆  2020   8   1   1.466667   0.900000  0.999745   
50   송파구  몽촌토성역 1번출구 옆  2020   8   2   1.266667   1.533333  0.817503   
51   송파구  몽촌토성역 1번출구 옆  2020   8   3   0.766667   1.100000  0.696027   
52   송파구  몽촌토성역 1번출구 옆  2020   8   4   0.633333   0.466667  0.997283   
53   송파구  몽촌토성역 1번출구 옆  2020   8   5   0.900000   0.400000  0.999997   
54   송파구  몽촌토성역 1번출구 옆  2020   8   6   0.9000

In [7]:
df_new['내 균형'] = '균형'
df_new

자치구          대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  몽촌토성역 1번출구 옆  2020   6   0   2.766667   3.000000  0.892904   
1    송파구  몽촌토성역 1번출구 옆  2020   6   1   2.166667   1.633333  0.996409   
2    송파구  몽촌토성역 1번출구 옆  2020   6   2   1.400000   1.866667  0.747469   
3    송파구  몽촌토성역 1번출구 옆  2020   6   3   1.100000   1.000000  0.972183   
4    송파구  몽촌토성역 1번출구 옆  2020   6   4   0.800000   0.966667  0.818813   
5    송파구  몽촌토성역 1번출구 옆  2020   6   5   0.433333   0.500000  0.851637   
6    송파구  몽촌토성역 1번출구 옆  2020   6   6   1.200000   1.133333  0.960669   
7    송파구  몽촌토성역 1번출구 옆  2020   6   7   2.900000   2.600000  0.975649   
8    송파구  몽촌토성역 1번출구 옆  2020   6   8   4.700000   3.733333  0.993329   
9    송파구  몽촌토성역 1번출구 옆  2020   6   9   2.533333   3.066667  0.817503   
10   송파구  몽촌토성역 1번출구 옆  2020   6  10   2.366667   2.200000  0.965826   
11   송파구  몽촌토성역 1번출구 옆  2020   6  11   3.733333   3.633333  0.949412   
12   송파구  몽촌토성역 1번출구 옆  2020   6  12   3.466667   3.166667  0.970900   
13   송파구  몽촌토성역 1번출구 옆  2020   6  13   4.366667   3.933333  0.974521   
14   송파구  몽촌토성역 1번출구 옆  2020   6  14   4.700000   4.466667  0.958493   
15   송파구  몽촌토성역 1번출구 옆  2020   6  15   6.133333   7.066667  0.852646   
16   송파구  몽촌토성역 1번출구 옆  2020   6  16   8.033333   8.100000  0.933552   
17   송파구  몽촌토성역 1번출구 옆  2020   6  17  11.000000  12.300000  0.873056   
18   송파구  몽촌토성역 1번출구 옆  2020   6  18  11.300000  11.766667  0.916793   
19   송파구  몽촌토성역 1번출구 옆  2020   6  19  11.733333  11.866667  0.932065   
20   송파구  몽촌토성역 1번출구 옆  2020   6  20  13.500000  13.000000  0.953613   
21   송파구  몽촌토성역 1번출구 옆  2020   6  21  11.933333  12.100000  0.930790   
22   송파구  몽촌토성역 1번출구 옆  2020   6  22   9.433333   9.166667  0.950032   
23   송파구  몽촌토성역 1번출구 옆  2020   6  23   6.033333   5.200000  0.983624   
24   송파구  몽촌토성역 1번출구 옆  2020   7   0   1.933333   1.766667  0.970801   
25   송파구  몽촌토성역 1번출구 옆  2020   7   1   1.700000   1.500000  0.979194   
26   송파구  몽촌토성역 1번출구 옆  2020   7   2   1.233333   1.633333  0.752335   
27   송파구  몽촌토성역 1번출구 옆  2020   7   3   1.066667   1.100000  0.922062   
28   송파구  몽촌토성역 1번출구 옆  2020   7   4   0.500000   0.733333  0.681123   
29   송파구  몽촌토성역 1번출구 옆  2020   7   5   0.433333   0.366667  0.986513   
30   송파구  몽촌토성역 1번출구 옆  2020   7   6   1.400000   0.866667  0.999714   
31   송파구  몽촌토성역 1번출구 옆  2020   7   7   2.533333   2.666667  0.910694   
32   송파구  몽촌토성역 1번출구 옆  2020   7   8   3.866667   2.166667  0.999926   
33   송파구  몽촌토성역 1번출구 옆  2020   7   9   1.966667   2.033333  0.920686   
34   송파구  몽촌토성역 1번출구 옆  2020   7  10   2.066667   2.033333  0.944831   
35   송파구  몽촌토성역 1번출구 옆  2020   7  11   2.166667   2.200000  0.930091   
36   송파구  몽촌토성역 1번출구 옆  2020   7  12   2.366667   2.733333  0.850982   
37   송파구  몽촌토성역 1번출구 옆  2020   7  13   3.833333   3.633333  0.959433   
38   송파구  몽촌토성역 1번출구 옆  2020   7  14   3.633333   4.033333  0.877859   
39   송파구  몽촌토성역 1번출구 옆  2020   7  15   4.200000   4.533333  0.895755   
40   송파구  몽촌토성역 1번출구 옆  2020   7  16   5.366667   5.233333  0.948594   
41   송파구  몽촌토성역 1번출구 옆  2020   7  17   7.333333   8.966667  0.810242   
42   송파구  몽촌토성역 1번출구 옆  2020   7  18   7.800000   7.966667  0.927107   
43   송파구  몽촌토성역 1번출구 옆  2020   7  19   8.833333   8.366667  0.959675   
44   송파구  몽촌토성역 1번출구 옆  2020   7  20   8.100000   8.133333  0.935557   
45   송파구  몽촌토성역 1번출구 옆  2020   7  21   8.266667   8.133333  0.944831   
46   송파구  몽촌토성역 1번출구 옆  2020   7  22   5.100000   5.166667  0.931229   
47   송파구  몽촌토성역 1번출구 옆  2020   7  23   3.433333   3.266667  0.958079   
48   송파구  몽촌토성역 1번출구 옆  2020   8   0   1.500000   1.700000  0.863995   
49   송파구  몽촌토성역 1번출구 옆  2020   8   1   1.466667   0.900000  0.999745   
50   송파구  몽촌토성역 1번출구 옆  2020   8   2   1.266667   1.533333  0.817503   
51   송파구  몽촌토성역 1번출구 옆  2020   8   3   0.766667   1.100000  0.696027   
52   송파구  몽촌토성역 1번출구 옆  2020   8   4   0.633333   0.466667  0.997283   
53   송파구  몽촌토성역 1번출구 옆  2020   8   5   0.900000   0.400000  0.999997   
54   송파구  몽촌토성역 1번출구 옆  2020   8   6   0.9000

# 일단 L_s, 기존균형 비교

## L_s가 Q1보다 작거나, Q3보다 크면 불균형

### L_s가 q1보다 작지만 이때 p_1+이 충분히 크다면 균형
- ls가 q1보다 작고 p_1+가 0.5보다 작으면 불균형

In [8]:
# ls가 q1보다 작으므로 불균형

e = df_new[(df_new['L_s'] < df_new['1사분위수']) & (df_new['p_1+'] < 0.6)]
e

자치구          대여소명     년   월  시간    lambda        mu      p_1+       L_s  \
149  송파구  몽촌토성역 1번출구 옆  2020  12   5  0.033333  0.100000  0.333333  0.500000   
170  송파구  몽촌토성역 1번출구 옆  2021   1   2  0.066667  0.466667  0.142857  0.166667   
172  송파구  몽촌토성역 1번출구 옆  2021   1   4  0.100000  0.000000  0.000000  0.000000   
173  송파구  몽촌토성역 1번출구 옆  2021   1   5  0.033333  0.133333  0.250000  0.333333   
191  송파구  몽촌토성역 1번출구 옆  2021   1  23  0.200000  0.466667  0.428571  0.749979   
196  송파구  몽촌토성역 1번출구 옆  2021   2   4  0.033333  0.000000  0.000000  0.000000   
197  송파구  몽촌토성역 1번출구 옆  2021   2   5  0.000000  0.300000  0.000000  0.000000   
198  송파구  몽촌토성역 1번출구 옆  2021   2   6  0.066667  0.900000  0.074074  0.080000   
218  송파구  몽촌토성역 1번출구 옆  2021   3   2  0.266667  1.100000  0.242424  0.320000   
221  송파구  몽촌토성역 1번출구 옆  2021   3   5  0.066667  0.166667  0.400000  0.666660   
242  송파구  몽촌토성역 1번출구 옆  2021   4   2  0.266667  0.966667  0.275862  0.380952   
245  송파구  몽촌토성역 1번출구 옆  2021   4   5  0.033333  0.133333  0.250000  0.333333   
246  송파구  몽촌토성역 1번출구 옆  2021   4   6  0.500000  1.000000  0.499992  0.999756   
268  송파구  몽촌토성역 1번출구 옆  2021   5   4  0.266667  0.466667  0.571373  1.331265   
270  송파구  몽촌토성역 1번출구 옆  2021   5   6  0.433333  0.966667  0.448274  0.812457   

           W_s        재고량  3사분위수  2사분위수     1사분위수   균형 내 균형  
149  14.999990   3.133333    8.7   5.55  3.733333  불균형   균형  
170   2.500000   5.833333    8.7   5.55  3.733333   균형   균형  
172   0.000000   6.066667    8.7   5.55  3.733333   균형   균형  
173  10.000000   6.100000    8.7   5.55  3.733333   균형   균형  
191   3.749903   5.633333    8.7   5.55  3.733333   균형   균형  
196   0.000000   8.266667    8.7   5.55  3.733333   균형   균형  
197   0.000000   8.266667    8.7   5.55  3.733333   균형   균형  
198   1.200000   8.266667    8.7   5.55  3.733333   균형   균형  
218   1.200000  12.100000    8.7   5.55  3.733333  불균형   균형  
221   9.999903  12.433333    8.7   5.55  3.733333  불균형   균형  
242   1.428571   6.600000    8.7   5.55  3.733333   균형   균형  
245  10.000000   7.366667    8.7   5.55  3.733333   균형   균형  
246   1.999542   7.366667    8.7   5.55  3.733333   균형   균형  
268   4.992729   9.033333    8.7   5.55  3.733333  불균형   균형  
270   1.874908   8.933333    8.7   5.55  3.733333  불균형   균형

In [9]:
for i in range(len(df_new)):
    if (df_new['L_s'][i] < df_new['1사분위수'][i]) & (df_new['p_1+'][i] < 0.6):
        df_new['내 균형'][i] = '불균형'

In [10]:
# ls가 q3보다 크므로 불균형

r = df_new[(df_new['L_s'] > df_new['3사분위수']) & (df_new['L_s'] > (15*0.7))]
r

자치구          대여소명     년   월  시간     lambda         mu      p_1+  \
1    송파구  몽촌토성역 1번출구 옆  2020   6   1   2.166667   1.633333  0.996409   
8    송파구  몽촌토성역 1번출구 옆  2020   6   8   4.700000   3.733333  0.993329   
29   송파구  몽촌토성역 1번출구 옆  2020   7   5   0.433333   0.366667  0.986513   
30   송파구  몽촌토성역 1번출구 옆  2020   7   6   1.400000   0.866667  0.999714   
32   송파구  몽촌토성역 1번출구 옆  2020   7   8   3.866667   2.166667  0.999926   
49   송파구  몽촌토성역 1번출구 옆  2020   8   1   1.466667   0.900000  0.999745   
52   송파구  몽촌토성역 1번출구 옆  2020   8   4   0.633333   0.466667  0.997283   
53   송파구  몽촌토성역 1번출구 옆  2020   8   5   0.900000   0.400000  0.999997   
54   송파구  몽촌토성역 1번출구 옆  2020   8   6   0.900000   0.700000  0.994782   
59   송파구  몽촌토성역 1번출구 옆  2020   8  11   2.200000   1.833333  0.988564   
61   송파구  몽촌토성역 1번출구 옆  2020   8  13   3.000000   2.366667  0.993839   
73   송파구  몽촌토성역 1번출구 옆  2020   9   1   1.766667   1.233333  0.998619   
78   송파구  몽촌토성역 1번출구 옆  2020   9   6   0.900000   0.466667  0.999975   
79   송파구  몽촌토성역 1번출구 옆  2020   9   7   2.966667   1.833333  0.999720   
81   송파구  몽촌토성역 1번출구 옆  2020   9   9   3.366667   2.866667  0.985581   
97   송파구  몽촌토성역 1번출구 옆  2020  10   1   1.566667   0.900000  0.999896   
100  송파구  몽촌토성역 1번출구 옆  2020  10   4   0.566667   0.233333  0.999999   
102  송파구  몽촌토성역 1번출구 옆  2020  10   6   1.300000   0.366667  1.000000   
105  송파구  몽촌토성역 1번출구 옆  2020  10   9   5.200000   3.800000  0.997547   
123  송파구  몽촌토성역 1번출구 옆  2020  11   3   0.533333   0.133333  1.000000   
125  송파구  몽촌토성역 1번출구 옆  2020  11   5   0.400000   0.266667  0.999238   
127  송파구  몽촌토성역 1번출구 옆  2020  11   7   1.366667   1.133333  0.989160   
128  송파구  몽촌토성역 1번출구 옆  2020  11   8   3.333333   2.000000  0.999812   
129  송파구  몽촌토성역 1번출구 옆  2020  11   9   3.666667   1.966667  0.999959   
130  송파구  몽촌토성역 1번출구 옆  2020  11  10   2.800000   2.266667  0.991715   
144  송파구  몽촌토성역 1번출구 옆  2020  12   0   0.533333   0.366667  0.998865   
145  송파구  몽촌토성역 1번출구 옆  2020  12   1   0.500000   0.200000  0.999999   
147  송파구  몽촌토성역 1번출구 옆  2020  12   3   0.166667   0.100000  0.999812   
148  송파구  몽촌토성역 1번출구 옆  2020  12   4   0.100000   0.033333  1.000000   
150  송파구  몽촌토성역 1번출구 옆  2020  12   6   0.466667   0.133333  1.000000   
151  송파구  몽촌토성역 1번출구 옆  2020  12   7   1.600000   0.933333  0.999872   
152  송파구  몽촌토성역 1번출구 옆  2020  12   8   1.633333   0.933333  0.999903   
153  송파구  몽촌토성역 1번출구 옆  2020  12   9   1.466667   1.000000  0.998980   
154  송파구  몽촌토성역 1번출구 옆  2020  12  10   1.600000   1.100000  0.998865   
157  송파구  몽촌토성역 1번출구 옆  2020  12  13   3.133333   2.233333  0.998204   
161  송파구  몽촌토성역 1번출구 옆  2020  12  17   4.233333   3.466667  0.990570   
168  송파구  몽촌토성역 1번출구 옆  2021   1   0   0.533333   0.266667  0.999985   
171  송파구  몽촌토성역 1번출구 옆  2021   1   3   0.133333   0.100000  0.996625   
174  송파구  몽촌토성역 1번출구 옆  2021   1   6   0.333333   0.166667  0.999985   
176  송파구  몽촌토성역 1번출구 옆  2021   1   8   2.233333   0.500000  1.000000   
177  송파구  몽촌토성역 1번출구 옆  2021   1   9   0.866667   0.566667  0.999409   
190  송파구  몽촌토성역 1번출구 옆  2021   1  22   0.900000   0.633333  0.998472   
193  송파구  몽촌토성역 1번출구 옆  2021   2   1   0.333333   0.200000  0.999812   
199  송파구  몽촌토성역 1번출구 옆  2021   2   7   0.833333   0.633333  0.996039   
200  송파구  몽촌토성역 1번출구 옆  2021   2   8   3.200000   0.666667  1.000000   
201  송파구  몽촌토성역 1번출구 옆  2021   2   9   2.766667   1.200000  0.999998   
204  송파구  몽촌토성역 1번출구 옆  2021   2  12   3.400000   2.800000  0.989960   
217  송파구  몽촌토성역 1번출구 옆  2021   3   1   0.733333   0.333333  0.999996   
219  송파구  몽촌토성역 1번출구 옆  2021   3   3   0.233333   0.100000  0.999998   
223  송파구  몽촌토성역 1번출구 옆  2021   3   7   2.166667   1.566667  0.997849   
224  송파구  몽촌토성역 1번출구 옆  2021   3   8   5.633333   2.500000  0.999997   
225  송파구  몽촌토성역 1번출구 옆  2021   3   9   5.866667   2.800000  0.999992   
228  송파구  몽촌토성역 1번출구 옆  2021   3  12   5.233333   4.000000  0.995759   
240  송파구  몽촌토성역 1번출구 옆  2021   4   0   1.733333   1.400000  0.991924   
243  송파구  몽촌토성역 1번출구 옆  2021   4   3   0.7666

In [11]:
for i in range(len(df_new)):
    if (df_new['L_s'][i] > df_new['3사분위수'][i]) & (df_new['L_s'][i] > (15*0.7)):
        df_new['내 균형'][i] = '불균형'

# L_s가 Q1~Q3사이이더라도 이때 p_1+가 너무 작으면(50%이하) 불균형
- 따릉이 목표 거치율이 60~70%이므로 한대라도 있을 확률인 p_1+도 유사하게 잡기 위해

In [12]:
t = df_new[(df_new['3사분위수']>=df_new['L_s']) & (df_new['L_s'] >= df_new['1사분위수'])]
t

자치구          대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  몽촌토성역 1번출구 옆  2020   6   0   2.766667   3.000000  0.892904   
4    송파구  몽촌토성역 1번출구 옆  2020   6   4   0.800000   0.966667  0.818813   
5    송파구  몽촌토성역 1번출구 옆  2020   6   5   0.433333   0.500000  0.851637   
6    송파구  몽촌토성역 1번출구 옆  2020   6   6   1.200000   1.133333  0.960669   
9    송파구  몽촌토성역 1번출구 옆  2020   6   9   2.533333   3.066667  0.817503   
11   송파구  몽촌토성역 1번출구 옆  2020   6  11   3.733333   3.633333  0.949412   
14   송파구  몽촌토성역 1번출구 옆  2020   6  14   4.700000   4.466667  0.958493   
15   송파구  몽촌토성역 1번출구 옆  2020   6  15   6.133333   7.066667  0.852646   
16   송파구  몽촌토성역 1번출구 옆  2020   6  16   8.033333   8.100000  0.933552   
17   송파구  몽촌토성역 1번출구 옆  2020   6  17  11.000000  12.300000  0.873056   
18   송파구  몽촌토성역 1번출구 옆  2020   6  18  11.300000  11.766667  0.916793   
19   송파구  몽촌토성역 1번출구 옆  2020   6  19  11.733333  11.866667  0.932065   
20   송파구  몽촌토성역 1번출구 옆  2020   6  20  13.500000  13.000000  0.953613   
21   송파구  몽촌토성역 1번출구 옆  2020   6  21  11.933333  12.100000  0.930790   
22   송파구  몽촌토성역 1번출구 옆  2020   6  22   9.433333   9.166667  0.950032   
27   송파구  몽촌토성역 1번출구 옆  2020   7   3   1.066667   1.100000  0.922062   
31   송파구  몽촌토성역 1번출구 옆  2020   7   7   2.533333   2.666667  0.910694   
33   송파구  몽촌토성역 1번출구 옆  2020   7   9   1.966667   2.033333  0.920686   
34   송파구  몽촌토성역 1번출구 옆  2020   7  10   2.066667   2.033333  0.944831   
35   송파구  몽촌토성역 1번출구 옆  2020   7  11   2.166667   2.200000  0.930091   
36   송파구  몽촌토성역 1번출구 옆  2020   7  12   2.366667   2.733333  0.850982   
37   송파구  몽촌토성역 1번출구 옆  2020   7  13   3.833333   3.633333  0.959433   
38   송파구  몽촌토성역 1번출구 옆  2020   7  14   3.633333   4.033333  0.877859   
39   송파구  몽촌토성역 1번출구 옆  2020   7  15   4.200000   4.533333  0.895755   
40   송파구  몽촌토성역 1번출구 옆  2020   7  16   5.366667   5.233333  0.948594   
41   송파구  몽촌토성역 1번출구 옆  2020   7  17   7.333333   8.966667  0.810242   
42   송파구  몽촌토성역 1번출구 옆  2020   7  18   7.800000   7.966667  0.927107   
43   송파구  몽촌토성역 1번출구 옆  2020   7  19   8.833333   8.366667  0.959675   
44   송파구  몽촌토성역 1번출구 옆  2020   7  20   8.100000   8.133333  0.935557   
45   송파구  몽촌토성역 1번출구 옆  2020   7  21   8.266667   8.133333  0.944831   
46   송파구  몽촌토성역 1번출구 옆  2020   7  22   5.100000   5.166667  0.931229   
47   송파구  몽촌토성역 1번출구 옆  2020   7  23   3.433333   3.266667  0.958079   
48   송파구  몽촌토성역 1번출구 옆  2020   8   0   1.500000   1.700000  0.863995   
50   송파구  몽촌토성역 1번출구 옆  2020   8   2   1.266667   1.533333  0.817503   
55   송파구  몽촌토성역 1번출구 옆  2020   8   7   1.900000   2.033333  0.900969   
63   송파구  몽촌토성역 1번출구 옆  2020   8  15   3.333333   3.733333  0.871973   
64   송파구  몽촌토성역 1번출구 옆  2020   8  16   3.666667   3.700000  0.933169   
65   송파구  몽촌토성역 1번출구 옆  2020   8  17   4.833333   5.566667  0.852918   
66   송파구  몽촌토성역 1번출구 옆  2020   8  18   5.966667   6.033333  0.932157   
68   송파구  몽촌토성역 1번출구 옆  2020   8  20   6.033333   6.200000  0.923929   
69   송파구  몽촌토성역 1번출구 옆  2020   8  21   4.366667   4.633333  0.906057   
70   송파구  몽촌토성역 1번출구 옆  2020   8  22   3.866667   3.766667  0.949024   
71   송파구  몽촌토성역 1번출구 옆  2020   8  23   2.300000   2.333333  0.930531   
72   송파구  몽촌토성역 1번출구 옆  2020   9   0   1.833333   2.033333  0.878451   
75   송파구  몽촌토성역 1번출구 옆  2020   9   3   0.966667   1.100000  0.861231   
76   송파구  몽촌토성역 1번출구 옆  2020   9   4   0.366667   0.400000  0.889106   
77   송파구  몽촌토성역 1번출구 옆  2020   9   5   0.533333   0.533333  0.937500   
82   송파구  몽촌토성역 1번출구 옆  2020   9  10   3.700000   4.133333  0.873690   
84   송파구  몽촌토성역 1번출구 옆  2020   9  12   6.600000   7.033333  0.903504   
85   송파구  몽촌토성역 1번출구 옆  2020   9  13   8.833333   8.966667  0.930234   
86   송파구  몽촌토성역 1번출구 옆  2020   9  14  10.533333  10.266667  0.948793   
87   송파구  몽촌토성역 1번출구 옆  2020   9  15  11.266667  11.133333  0.942925   
88   송파구  몽촌토성역 1번출구 옆  2020   9  16  13.433333  14.200000  0.908264   
89   송파구  몽촌토성역 1번출구 옆  2020   9  17  16.800000  17.066667  0.929849   
90   송파구  몽촌토성역 1번출구 옆  2020   9  18  17.2333

In [13]:
t[t['p_1+']< 0.6]

Empty DataFrame
Columns: [자치구, 대여소명, 년, 월, 시간, lambda, mu, p_1+, L_s, W_s, 재고량, 3사분위수, 2사분위수, 1사분위수, 균형, 내 균형]
Index: []

- 위 결과를 보면 l_s가 q1~q3사이에 있지만 자전거가 한대라도 있을 확률이 0.5이하이므로 불균형이다
- 특히 19시, 20시의 경우 퇴근시간이므로 이시간에 불균형이면 이용에 불편을 초래함
- 또한 21년 1월 19시의 경우 불균형인데도 불구하고 기존연구방법에서는 이를 균형으로 판단

In [14]:
for i in range(len(df_new)):
    if (df_new['3사분위수'][i]>=df_new['L_s'][i]) & (df_new['L_s'][i] >= df_new['1사분위수'][i]) & (df_new['p_1+'][i] < 0.6):
        df_new['내 균형'][i] = '불균형'

In [15]:
df_new

자치구          대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  몽촌토성역 1번출구 옆  2020   6   0   2.766667   3.000000  0.892904   
1    송파구  몽촌토성역 1번출구 옆  2020   6   1   2.166667   1.633333  0.996409   
2    송파구  몽촌토성역 1번출구 옆  2020   6   2   1.400000   1.866667  0.747469   
3    송파구  몽촌토성역 1번출구 옆  2020   6   3   1.100000   1.000000  0.972183   
4    송파구  몽촌토성역 1번출구 옆  2020   6   4   0.800000   0.966667  0.818813   
5    송파구  몽촌토성역 1번출구 옆  2020   6   5   0.433333   0.500000  0.851637   
6    송파구  몽촌토성역 1번출구 옆  2020   6   6   1.200000   1.133333  0.960669   
7    송파구  몽촌토성역 1번출구 옆  2020   6   7   2.900000   2.600000  0.975649   
8    송파구  몽촌토성역 1번출구 옆  2020   6   8   4.700000   3.733333  0.993329   
9    송파구  몽촌토성역 1번출구 옆  2020   6   9   2.533333   3.066667  0.817503   
10   송파구  몽촌토성역 1번출구 옆  2020   6  10   2.366667   2.200000  0.965826   
11   송파구  몽촌토성역 1번출구 옆  2020   6  11   3.733333   3.633333  0.949412   
12   송파구  몽촌토성역 1번출구 옆  2020   6  12   3.466667   3.166667  0.970900   
13   송파구  몽촌토성역 1번출구 옆  2020   6  13   4.366667   3.933333  0.974521   
14   송파구  몽촌토성역 1번출구 옆  2020   6  14   4.700000   4.466667  0.958493   
15   송파구  몽촌토성역 1번출구 옆  2020   6  15   6.133333   7.066667  0.852646   
16   송파구  몽촌토성역 1번출구 옆  2020   6  16   8.033333   8.100000  0.933552   
17   송파구  몽촌토성역 1번출구 옆  2020   6  17  11.000000  12.300000  0.873056   
18   송파구  몽촌토성역 1번출구 옆  2020   6  18  11.300000  11.766667  0.916793   
19   송파구  몽촌토성역 1번출구 옆  2020   6  19  11.733333  11.866667  0.932065   
20   송파구  몽촌토성역 1번출구 옆  2020   6  20  13.500000  13.000000  0.953613   
21   송파구  몽촌토성역 1번출구 옆  2020   6  21  11.933333  12.100000  0.930790   
22   송파구  몽촌토성역 1번출구 옆  2020   6  22   9.433333   9.166667  0.950032   
23   송파구  몽촌토성역 1번출구 옆  2020   6  23   6.033333   5.200000  0.983624   
24   송파구  몽촌토성역 1번출구 옆  2020   7   0   1.933333   1.766667  0.970801   
25   송파구  몽촌토성역 1번출구 옆  2020   7   1   1.700000   1.500000  0.979194   
26   송파구  몽촌토성역 1번출구 옆  2020   7   2   1.233333   1.633333  0.752335   
27   송파구  몽촌토성역 1번출구 옆  2020   7   3   1.066667   1.100000  0.922062   
28   송파구  몽촌토성역 1번출구 옆  2020   7   4   0.500000   0.733333  0.681123   
29   송파구  몽촌토성역 1번출구 옆  2020   7   5   0.433333   0.366667  0.986513   
30   송파구  몽촌토성역 1번출구 옆  2020   7   6   1.400000   0.866667  0.999714   
31   송파구  몽촌토성역 1번출구 옆  2020   7   7   2.533333   2.666667  0.910694   
32   송파구  몽촌토성역 1번출구 옆  2020   7   8   3.866667   2.166667  0.999926   
33   송파구  몽촌토성역 1번출구 옆  2020   7   9   1.966667   2.033333  0.920686   
34   송파구  몽촌토성역 1번출구 옆  2020   7  10   2.066667   2.033333  0.944831   
35   송파구  몽촌토성역 1번출구 옆  2020   7  11   2.166667   2.200000  0.930091   
36   송파구  몽촌토성역 1번출구 옆  2020   7  12   2.366667   2.733333  0.850982   
37   송파구  몽촌토성역 1번출구 옆  2020   7  13   3.833333   3.633333  0.959433   
38   송파구  몽촌토성역 1번출구 옆  2020   7  14   3.633333   4.033333  0.877859   
39   송파구  몽촌토성역 1번출구 옆  2020   7  15   4.200000   4.533333  0.895755   
40   송파구  몽촌토성역 1번출구 옆  2020   7  16   5.366667   5.233333  0.948594   
41   송파구  몽촌토성역 1번출구 옆  2020   7  17   7.333333   8.966667  0.810242   
42   송파구  몽촌토성역 1번출구 옆  2020   7  18   7.800000   7.966667  0.927107   
43   송파구  몽촌토성역 1번출구 옆  2020   7  19   8.833333   8.366667  0.959675   
44   송파구  몽촌토성역 1번출구 옆  2020   7  20   8.100000   8.133333  0.935557   
45   송파구  몽촌토성역 1번출구 옆  2020   7  21   8.266667   8.133333  0.944831   
46   송파구  몽촌토성역 1번출구 옆  2020   7  22   5.100000   5.166667  0.931229   
47   송파구  몽촌토성역 1번출구 옆  2020   7  23   3.433333   3.266667  0.958079   
48   송파구  몽촌토성역 1번출구 옆  2020   8   0   1.500000   1.700000  0.863995   
49   송파구  몽촌토성역 1번출구 옆  2020   8   1   1.466667   0.900000  0.999745   
50   송파구  몽촌토성역 1번출구 옆  2020   8   2   1.266667   1.533333  0.817503   
51   송파구  몽촌토성역 1번출구 옆  2020   8   3   0.766667   1.100000  0.696027   
52   송파구  몽촌토성역 1번출구 옆  2020   8   4   0.633333   0.466667  0.997283   
53   송파구  몽촌토성역 1번출구 옆  2020   8   5   0.900000   0.400000  0.999997   
54   송파구  몽촌토성역 1번출구 옆  2020   8   6   0.9000

In [16]:
len(df_new[df_new['균형'] == '불균형'])

142

In [17]:
len(df_new[df_new['내 균형'] == '불균형'])

80

In [18]:
len(df_new)

288

In [19]:
# TN

len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '불균형')])

39

In [20]:
# TP

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')])

105

In [21]:
# FN

len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')])

103

In [22]:
# FP

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')])

41

In [23]:
# 정확도(기존방법과 내 방법의 일치율)

(len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '불균형')])) / len(df_new)

0.5

In [24]:
# 정밀도(기존 방법에서 균형으로 판단한 데이터 중 내가 균형이라고 판단한 비율)

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) / (len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')]))

0.7191780821917808

In [25]:
# 재현율(내 방법에서 균형으로 판단한 것 중 기존 방법에서도 균형으로 판단한 비율)

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) / (len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')]))

0.5048076923076923

In [26]:
a = df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')]
a

자치구          대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  몽촌토성역 1번출구 옆  2020   6   0   2.766667   3.000000  0.892904   
2    송파구  몽촌토성역 1번출구 옆  2020   6   2   1.400000   1.866667  0.747469   
9    송파구  몽촌토성역 1번출구 옆  2020   6   9   2.533333   3.066667  0.817503   
10   송파구  몽촌토성역 1번출구 옆  2020   6  10   2.366667   2.200000  0.965826   
11   송파구  몽촌토성역 1번출구 옆  2020   6  11   3.733333   3.633333  0.949412   
15   송파구  몽촌토성역 1번출구 옆  2020   6  15   6.133333   7.066667  0.852646   
17   송파구  몽촌토성역 1번출구 옆  2020   6  17  11.000000  12.300000  0.873056   
18   송파구  몽촌토성역 1번출구 옆  2020   6  18  11.300000  11.766667  0.916793   
19   송파구  몽촌토성역 1번출구 옆  2020   6  19  11.733333  11.866667  0.932065   
20   송파구  몽촌토성역 1번출구 옆  2020   6  20  13.500000  13.000000  0.953613   
21   송파구  몽촌토성역 1번출구 옆  2020   6  21  11.933333  12.100000  0.930790   
22   송파구  몽촌토성역 1번출구 옆  2020   6  22   9.433333   9.166667  0.950032   
23   송파구  몽촌토성역 1번출구 옆  2020   6  23   6.033333   5.200000  0.983624   
24   송파구  몽촌토성역 1번출구 옆  2020   7   0   1.933333   1.766667  0.970801   
25   송파구  몽촌토성역 1번출구 옆  2020   7   1   1.700000   1.500000  0.979194   
26   송파구  몽촌토성역 1번출구 옆  2020   7   2   1.233333   1.633333  0.752335   
27   송파구  몽촌토성역 1번출구 옆  2020   7   3   1.066667   1.100000  0.922062   
28   송파구  몽촌토성역 1번출구 옆  2020   7   4   0.500000   0.733333  0.681123   
31   송파구  몽촌토성역 1번출구 옆  2020   7   7   2.533333   2.666667  0.910694   
33   송파구  몽촌토성역 1번출구 옆  2020   7   9   1.966667   2.033333  0.920686   
34   송파구  몽촌토성역 1번출구 옆  2020   7  10   2.066667   2.033333  0.944831   
35   송파구  몽촌토성역 1번출구 옆  2020   7  11   2.166667   2.200000  0.930091   
36   송파구  몽촌토성역 1번출구 옆  2020   7  12   2.366667   2.733333  0.850982   
37   송파구  몽촌토성역 1번출구 옆  2020   7  13   3.833333   3.633333  0.959433   
38   송파구  몽촌토성역 1번출구 옆  2020   7  14   3.633333   4.033333  0.877859   
39   송파구  몽촌토성역 1번출구 옆  2020   7  15   4.200000   4.533333  0.895755   
40   송파구  몽촌토성역 1번출구 옆  2020   7  16   5.366667   5.233333  0.948594   
41   송파구  몽촌토성역 1번출구 옆  2020   7  17   7.333333   8.966667  0.810242   
42   송파구  몽촌토성역 1번출구 옆  2020   7  18   7.800000   7.966667  0.927107   
43   송파구  몽촌토성역 1번출구 옆  2020   7  19   8.833333   8.366667  0.959675   
44   송파구  몽촌토성역 1번출구 옆  2020   7  20   8.100000   8.133333  0.935557   
45   송파구  몽촌토성역 1번출구 옆  2020   7  21   8.266667   8.133333  0.944831   
46   송파구  몽촌토성역 1번출구 옆  2020   7  22   5.100000   5.166667  0.931229   
47   송파구  몽촌토성역 1번출구 옆  2020   7  23   3.433333   3.266667  0.958079   
56   송파구  몽촌토성역 1번출구 옆  2020   8   8   2.633333   2.333333  0.978303   
57   송파구  몽촌토성역 1번출구 옆  2020   8   9   1.766667   1.533333  0.982397   
58   송파구  몽촌토성역 1번출구 옆  2020   8  10   1.600000   2.166667  0.736400   
94   송파구  몽촌토성역 1번출구 옆  2020   9  22   8.600000  10.333333  0.822874   
118  송파구  몽촌토성역 1번출구 옆  2020  10  22   6.066667   7.600000  0.792610   
119  송파구  몽촌토성역 1번출구 옆  2020  10  23   4.166667   4.400000  0.908852   
120  송파구  몽촌토성역 1번출구 옆  2020  11   0   1.033333   0.966667  0.963832   
121  송파구  몽촌토성역 1번출구 옆  2020  11   1   0.933333   1.233333  0.753910   
122  송파구  몽촌토성역 1번출구 옆  2020  11   2   0.866667   1.033333  0.828424   
124  송파구  몽촌토성역 1번출구 옆  2020  11   4   0.333333   0.333333  0.937500   
126  송파구  몽촌토성역 1번출구 옆  2020  11   6   0.700000   0.700000  0.937500   
131  송파구  몽촌토성역 1번출구 옆  2020  11  11   3.866667   3.500000  0.973301   
132  송파구  몽촌토성역 1번출구 옆  2020  11  12   4.933333   4.866667  0.943674   
133  송파구  몽촌토성역 1번출구 옆  2020  11  13   5.000000   4.666667  0.964567   
134  송파구  몽촌토성역 1번출구 옆  2020  11  14   6.800000   7.433333  0.887811   
135  송파구  몽촌토성역 1번출구 옆  2020  11  15   8.533333   7.900000  0.967070   
136  송파구  몽촌토성역 1번출구 옆  2020  11  16   8.433333   8.300000  0.944690   
137  송파구  몽촌토성역 1번출구 옆  2020  11  17   8.133333   7.533333  0.966925   
138  송파구  몽촌토성역 1번출구 옆  2020  11  18   6.066667   8.000000  0.755408   
139  송파구  몽촌토성역 1번출구 옆  2020  11  19   4.533333   5.066667  0.873375   
140  송파구  몽촌토성역 1번출구 옆  2020  11  20   4.3666

In [27]:
len(a)

103

In [28]:
len(a[a['p_1+'] > 0.9])

58

In [29]:
len(a[a['p_1+'] > 0.8])

82

In [30]:
len(a[a['p_1+'] > 0.7])

94

In [31]:
len(a[a['p_1+'] > 0.6])

103

- 기존방법이 불균형으로 판단한 데이터 중 본 연구에서 균형으로 판단한 데이터는 103개이지만
- 이 중 대여소에 적어도 자전거가 1대 있을 확률이 70%가 넘는 데이터가 94개로 90%이상의 경우에 이용자가 대여소에 도착했을 때 자전거를 빌릴 수 있는 확률이 70%이상이었다.
- 따라서 이러한 경우에는 자전거 재배치를 위한 대여소 내 재고의 균형/ 불균형을 판단할 경우에는 재배치를 할 필요가 없다고 판단하여 균형으로 판단할 수 있다.

In [32]:
b = df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')]
b

자치구          대여소명     년   월  시간    lambda        mu      p_1+        L_s  \
1    송파구  몽촌토성역 1번출구 옆  2020   6   1  2.166667  1.633333  0.996409  12.113452   
49   송파구  몽촌토성역 1번출구 옆  2020   8   1  1.466667  0.900000  0.999745  13.418234   
52   송파구  몽촌토성역 1번출구 옆  2020   8   4  0.633333  0.466667  0.997283  12.321731   
53   송파구  몽촌토성역 1번출구 옆  2020   8   5  0.900000  0.400000  0.999997  14.200037   
54   송파구  몽촌토성역 1번출구 옆  2020   8   6  0.900000  0.700000  0.994782  11.792192   
61   송파구  몽촌토성역 1번출구 옆  2020   8  13  3.000000  2.366667  0.993839  11.631506   
73   송파구  몽촌토성역 1번출구 옆  2020   9   1  1.766667  1.233333  0.998619  12.738588   
78   송파구  몽촌토성역 1번출구 옆  2020   9   6  0.900000  0.466667  0.999975  13.923514   
79   송파구  몽촌토성역 1번출구 옆  2020   9   7  2.966667  1.833333  0.999720  13.389595   
81   송파구  몽촌토성역 1번출구 옆  2020   9   9  3.366667  2.866667  0.985581  10.589325   
97   송파구  몽촌토성역 1번출구 옆  2020  10   1  1.566667  0.900000  0.999896  13.652251   
100  송파구  몽촌토성역 1번출구 옆  2020  10   4  0.566667  0.233333  0.999999  14.300011   
102  송파구  몽촌토성역 1번출구 옆  2020  10   6  1.300000  0.366667  1.000000  14.607143   
105  송파구  몽촌토성역 1번출구 옆  2020  10   9  5.200000  3.800000  0.997547  12.392248   
128  송파구  몽촌토성역 1번출구 옆  2020  11   8  3.333333  2.000000  0.999812  13.504515   
151  송파구  몽촌토성역 1번출구 옆  2020  12   7  1.600000  0.933333  0.999872  13.602877   
168  송파구  몽촌토성역 1번출구 옆  2021   1   0  0.533333  0.266667  0.999985  14.000244   
170  송파구  몽촌토성역 1번출구 옆  2021   1   2  0.066667  0.466667  0.142857   0.166667   
171  송파구  몽촌토성역 1번출구 옆  2021   1   3  0.133333  0.100000  0.996625  12.161985   
172  송파구  몽촌토성역 1번출구 옆  2021   1   4  0.100000  0.000000  0.000000   0.000000   
173  송파구  몽촌토성역 1번출구 옆  2021   1   5  0.033333  0.133333  0.250000   0.333333   
174  송파구  몽촌토성역 1번출구 옆  2021   1   6  0.333333  0.166667  0.999985  14.000244   
190  송파구  몽촌토성역 1번출구 옆  2021   1  22  0.900000  0.633333  0.998472  12.683067   
191  송파구  몽촌토성역 1번출구 옆  2021   1  23  0.200000  0.466667  0.428571   0.749979   
193  송파구  몽촌토성역 1번출구 옆  2021   2   1  0.333333  0.200000  0.999812  13.504515   
196  송파구  몽촌토성역 1번출구 옆  2021   2   4  0.033333  0.000000  0.000000   0.000000   
197  송파구  몽촌토성역 1번출구 옆  2021   2   5  0.000000  0.300000  0.000000   0.000000   
198  송파구  몽촌토성역 1번출구 옆  2021   2   6  0.066667  0.900000  0.074074   0.080000   
199  송파구  몽촌토성역 1번출구 옆  2021   2   7  0.833333  0.633333  0.996039  12.034035   
200  송파구  몽촌토성역 1번출구 옆  2021   2   8  3.200000  0.666667  1.000000  14.736842   
201  송파구  몽촌토성역 1번출구 옆  2021   2   9  2.766667  1.200000  0.999998  14.234068   
204  송파구  몽촌토성역 1번출구 옆  2021   2  12  3.400000  2.800000  0.989960  11.083007   
240  송파구  몽촌토성역 1번출구 옆  2021   4   0  1.733333  1.400000  0.991924  11.342676   
242  송파구  몽촌토성역 1번출구 옆  2021   4   2  0.266667  0.966667  0.275862   0.380952   
243  송파구  몽촌토성역 1번출구 옆  2021   4   3  0.766667  0.233333  1.000000  14.562500   
245  송파구  몽촌토성역 1번출구 옆  2021   4   5  0.033333  0.133333  0.250000   0.333333   
246  송파구  몽촌토성역 1번출구 옆  2021   4   6  0.500000  1.000000  0.499992   0.999756   
247  송파구  몽촌토성역 1번출구 옆  2021   4   7  3.100000  2.233333  0.997946  12.507777   
248  송파구  몽촌토성역 1번출구 옆  2021   4   8  9.066667  5.466667  0.999799  13.486364   
263  송파구  몽촌토성역 1번출구 옆  2021   4  23  4.533333  3.800000  0.987811  10.828771   
272  송파구  몽촌토성역 1번출구 옆  2021   5   8  6.833333  4.500000  0.999350  13.091471   

            W_s       재고량  3사분위수  2사분위수     1사분위수  균형 내 균형  
1      7.443126  4.033333    8.7   5.55  3.733333  균형  불균형  
49    14.912946  5.066667    8.7   5.55  3.733333  균형  불균형  
52    26.475650  5.700000    8.7   5.55  3.733333  균형  불균형  
53    35.500196  6.200000    8.7   5.55  3.733333  균형  불균형  
54    16.934347  6.400000    8.7   5.55  3.733333  균형  불균형  
61     4.945187  4.733333    8.7   5.55  3.733333  균형  불균형  
73    10.342866  4.566667    8.7   5.55  3.733333  균형  불균형  
78    29.836858  5.800000    8.7   5.55  3.733333  균형  불균형  
79     7.305460  6.933333    8.7   5.55  3.733333  균형

In [33]:
len(b)

41

In [34]:
len(b[(b['lambda'] > b['mu']) & (b['L_s'] > (15 * 0.7))])

31

In [35]:
31/41

0.7560975609756098

- 기존방법이 균형으로 판단한 데이터 중 본 연구에서 불균형으로 판단한 데이터는 41개이지만
- 이 중 반납이 대여보다 많고 대여소 내 평균 자전거수(L_s)가 서울시 거치대율의 목표인 거치대수의 60~70%를 초과하는 경우가 31개로 75%이상의 경우에 불필요하게 자전거의 재고가 많다고 판단된다.
- 따라서 이러한 경우에는 이용자들의 자전거 이용에는 문제가 없지만 주변 대여소의 재고가 부족하다면 자전거 재배치의 측면에서는 재배치를 할 필요가 있다고 판단하여 불균형으로 판단할 수 있다.

In [36]:
df_new

자치구          대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  몽촌토성역 1번출구 옆  2020   6   0   2.766667   3.000000  0.892904   
1    송파구  몽촌토성역 1번출구 옆  2020   6   1   2.166667   1.633333  0.996409   
2    송파구  몽촌토성역 1번출구 옆  2020   6   2   1.400000   1.866667  0.747469   
3    송파구  몽촌토성역 1번출구 옆  2020   6   3   1.100000   1.000000  0.972183   
4    송파구  몽촌토성역 1번출구 옆  2020   6   4   0.800000   0.966667  0.818813   
5    송파구  몽촌토성역 1번출구 옆  2020   6   5   0.433333   0.500000  0.851637   
6    송파구  몽촌토성역 1번출구 옆  2020   6   6   1.200000   1.133333  0.960669   
7    송파구  몽촌토성역 1번출구 옆  2020   6   7   2.900000   2.600000  0.975649   
8    송파구  몽촌토성역 1번출구 옆  2020   6   8   4.700000   3.733333  0.993329   
9    송파구  몽촌토성역 1번출구 옆  2020   6   9   2.533333   3.066667  0.817503   
10   송파구  몽촌토성역 1번출구 옆  2020   6  10   2.366667   2.200000  0.965826   
11   송파구  몽촌토성역 1번출구 옆  2020   6  11   3.733333   3.633333  0.949412   
12   송파구  몽촌토성역 1번출구 옆  2020   6  12   3.466667   3.166667  0.970900   
13   송파구  몽촌토성역 1번출구 옆  2020   6  13   4.366667   3.933333  0.974521   
14   송파구  몽촌토성역 1번출구 옆  2020   6  14   4.700000   4.466667  0.958493   
15   송파구  몽촌토성역 1번출구 옆  2020   6  15   6.133333   7.066667  0.852646   
16   송파구  몽촌토성역 1번출구 옆  2020   6  16   8.033333   8.100000  0.933552   
17   송파구  몽촌토성역 1번출구 옆  2020   6  17  11.000000  12.300000  0.873056   
18   송파구  몽촌토성역 1번출구 옆  2020   6  18  11.300000  11.766667  0.916793   
19   송파구  몽촌토성역 1번출구 옆  2020   6  19  11.733333  11.866667  0.932065   
20   송파구  몽촌토성역 1번출구 옆  2020   6  20  13.500000  13.000000  0.953613   
21   송파구  몽촌토성역 1번출구 옆  2020   6  21  11.933333  12.100000  0.930790   
22   송파구  몽촌토성역 1번출구 옆  2020   6  22   9.433333   9.166667  0.950032   
23   송파구  몽촌토성역 1번출구 옆  2020   6  23   6.033333   5.200000  0.983624   
24   송파구  몽촌토성역 1번출구 옆  2020   7   0   1.933333   1.766667  0.970801   
25   송파구  몽촌토성역 1번출구 옆  2020   7   1   1.700000   1.500000  0.979194   
26   송파구  몽촌토성역 1번출구 옆  2020   7   2   1.233333   1.633333  0.752335   
27   송파구  몽촌토성역 1번출구 옆  2020   7   3   1.066667   1.100000  0.922062   
28   송파구  몽촌토성역 1번출구 옆  2020   7   4   0.500000   0.733333  0.681123   
29   송파구  몽촌토성역 1번출구 옆  2020   7   5   0.433333   0.366667  0.986513   
30   송파구  몽촌토성역 1번출구 옆  2020   7   6   1.400000   0.866667  0.999714   
31   송파구  몽촌토성역 1번출구 옆  2020   7   7   2.533333   2.666667  0.910694   
32   송파구  몽촌토성역 1번출구 옆  2020   7   8   3.866667   2.166667  0.999926   
33   송파구  몽촌토성역 1번출구 옆  2020   7   9   1.966667   2.033333  0.920686   
34   송파구  몽촌토성역 1번출구 옆  2020   7  10   2.066667   2.033333  0.944831   
35   송파구  몽촌토성역 1번출구 옆  2020   7  11   2.166667   2.200000  0.930091   
36   송파구  몽촌토성역 1번출구 옆  2020   7  12   2.366667   2.733333  0.850982   
37   송파구  몽촌토성역 1번출구 옆  2020   7  13   3.833333   3.633333  0.959433   
38   송파구  몽촌토성역 1번출구 옆  2020   7  14   3.633333   4.033333  0.877859   
39   송파구  몽촌토성역 1번출구 옆  2020   7  15   4.200000   4.533333  0.895755   
40   송파구  몽촌토성역 1번출구 옆  2020   7  16   5.366667   5.233333  0.948594   
41   송파구  몽촌토성역 1번출구 옆  2020   7  17   7.333333   8.966667  0.810242   
42   송파구  몽촌토성역 1번출구 옆  2020   7  18   7.800000   7.966667  0.927107   
43   송파구  몽촌토성역 1번출구 옆  2020   7  19   8.833333   8.366667  0.959675   
44   송파구  몽촌토성역 1번출구 옆  2020   7  20   8.100000   8.133333  0.935557   
45   송파구  몽촌토성역 1번출구 옆  2020   7  21   8.266667   8.133333  0.944831   
46   송파구  몽촌토성역 1번출구 옆  2020   7  22   5.100000   5.166667  0.931229   
47   송파구  몽촌토성역 1번출구 옆  2020   7  23   3.433333   3.266667  0.958079   
48   송파구  몽촌토성역 1번출구 옆  2020   8   0   1.500000   1.700000  0.863995   
49   송파구  몽촌토성역 1번출구 옆  2020   8   1   1.466667   0.900000  0.999745   
50   송파구  몽촌토성역 1번출구 옆  2020   8   2   1.266667   1.533333  0.817503   
51   송파구  몽촌토성역 1번출구 옆  2020   8   3   0.766667   1.100000  0.696027   
52   송파구  몽촌토성역 1번출구 옆  2020   8   4   0.633333   0.466667  0.997283   
53   송파구  몽촌토성역 1번출구 옆  2020   8   5   0.900000   0.400000  0.999997   
54   송파구  몽촌토성역 1번출구 옆  2020   8   6   0.9000

In [37]:
df_new.to_csv('./몽촌토성역 1번출구 옆.csv', index = False, encoding = 'cp949')

# confusion matrix

In [23]:
len(df_new)

288

In [24]:
# 재고량이 ls보다 많거나 같을 경우 불균형 아님 = 균형임

z = df_new[df_new['L_s'] <= df_new['재고량']]
z

자치구               대여소명     년   월  시간     lambda         mu      p_1+  \
1    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   1   1.033333   1.300000  0.794608   
2    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   2   0.500000   0.666667  0.749940   
11   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  11  10.333333   8.333333  0.999530   
12   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  12  12.266667   8.766667  0.999977   
13   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  13  11.433333  12.033333  0.935505   
14   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  14  13.566667  15.266667  0.884895   
15   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  15  12.866667  17.133333  0.750911   
16   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  16  16.500000  20.166667  0.817640   
17   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  17  19.266667  28.900000  0.666664   
35   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  11   7.533333   7.000000  0.989718   
37   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  13  11.233333  10.133333  0.994244   
38   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  14  11.000000  15.633333  0.703614   
39   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  15  12.700000  15.433333  0.822271   
40   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  16  13.166667  19.166667  0.686951   
41   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  17  17.966667  23.166667  0.775398   
50   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   2   0.233333   0.433333  0.538462   
51   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   3   0.200000   0.233333  0.855489   
56   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   8   6.666667   8.733333  0.763265   
62   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  14   5.566667   7.033333  0.791232   
63   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  15   6.466667   7.500000  0.860325   
64   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  16   6.433333   8.366667  0.768811   
65   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  17   9.566667  11.500000  0.831072   
74   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9   2   0.233333   0.433333  0.538462   
85   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  13  14.566667  14.300000  0.973692   
86   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  14  14.100000  18.833333  0.748616   
87   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  15  16.600000  18.833333  0.878286   
88   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  16  18.200000  21.666667  0.838974   
89   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  17  19.700000  25.366667  0.776463   
108  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  12  14.633333  13.066667  0.995331   
109  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  13  15.866667  16.700000  0.935477   
110  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  14  16.866667  21.466667  0.785517   
111  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  15  18.700000  19.600000  0.938294   
112  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  16  22.033333  26.733333  0.823542   
120  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   0   0.700000   0.500000  0.999977   
121  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   1   1.100000   0.833333  0.999898   
122  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   2   0.466667   0.466667  0.965517   
123  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   3   0.033333   0.233333  0.142857   
124  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   4   0.266667   0.300000  0.885115   
126  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   6   1.966667   0.333333  1.000000   
127  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   7   8.833333   3.500000  1.000000   
128  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   8  16.366667   9.500000  1.000000   
129  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   9  10.900000   6.900000  0.999999   
130  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  10   8.433333   4.100000  1.000000   
131  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  11   6.733333   4.966667  0.999948   
132  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  12   9.200000   8.600000  0.988505   
133  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  13   9.800000   9.633333  0.973156   
134  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  14  11.066667  11.700000  0.932411   
135  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  15  13.500000  12.000000  0.995754   
136  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  16  13.666667  17.500000  0.780784   
137  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  17  14.700000  21.833333  0.673279   
138  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  18  16.400000  25.033333  0.655125   
1

In [25]:
len(z)

89

In [26]:
# TP

len(z[z['균형'] == '균형'])

34

In [27]:
# FN

len(z[z['균형'] == '불균형'])

55

- 재고량이 ls보다 많거나 같을 경우 균형인데 불균형이라고 판단한 수가 55개

- 균형인데 균형이라고 판단한 수가 34개

In [28]:
# ls보다 재고량이 작으므로 불균형

w = df_new[df_new['L_s'] > df_new['재고량']]
w

자치구               대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   0   1.666667   1.533333  0.991495   
3    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   3   0.366667   0.333333  0.993272   
4    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   4   0.833333   0.466667  1.000000   
5    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   5   1.200000   0.300000  1.000000   
6    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   6   2.933333   1.266667  1.000000   
7    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   7  10.200000   5.666667  1.000000   
8    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   8  19.033333  17.500000  0.991595   
9    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   9  17.633333   9.533333  1.000000   
10   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  10   9.733333   5.800000  1.000000   
18   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  18  27.400000  30.466667  0.894477   
19   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  19  17.966667  18.433333  0.951743   
20   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  20  14.533333  14.766667  0.957284   
21   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  21  11.633333  11.666667  0.964118   
22   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  22   7.933333   7.800000  0.973073   
23   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  23   2.966667   3.266667  0.902176   
24   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   0   1.733333   1.900000  0.905701   
25   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   1   0.900000   0.766667  0.998321   
26   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   2   0.433333   0.500000  0.864531   
27   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   3   0.633333   0.400000  0.999999   
28   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   4   0.766667   0.200000  1.000000   
29   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   5   1.266667   0.800000  0.999999   
30   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   6   2.533333   1.566667  0.999999   
31   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   7  10.033333   4.233333  1.000000   
32   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   8  15.533333  13.733333  0.996209   
33   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   9  13.333333   6.400000  1.000000   
34   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  10   7.800000   5.266667  0.999995   
36   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  12   9.700000   7.600000  0.999766   
42   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  18  21.366667  22.800000  0.925854   
43   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  19  15.400000  15.066667  0.975037   
44   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  20  11.600000  11.700000  0.961214   
45   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  21   8.366667   8.666667  0.945913   
46   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  22   5.700000   6.033333  0.931589   
47   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  23   2.800000   2.900000  0.945999   
48   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   0   1.333333   1.233333  0.990563   
49   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   1   0.900000   0.866667  0.980649   
52   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   4   0.166667   0.066667  1.000000   
53   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   5   0.733333   0.300000  1.000000   
54   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   6   1.400000   0.566667  1.000000   
55   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   7   5.866667   3.433333  1.000000   
57   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   9   7.766667   3.900000  1.000000   
58   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  10   5.433333   4.166667  0.999862   
59   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  11   5.566667   5.100000  0.992158   
60   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  12   5.266667   4.366667  0.999097   
61   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  13   5.933333   5.133333  0.997628   
66   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  18  10.333333  10.700000  0.946138   
67   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  19   8.800000   8.466667  0.980928   
68   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  20   6.733333   7.233333  0.920976   
69   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  21   5.100000   5.633333  0.899721   
70   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  22   3.733333   3.700000  0.969672   
71   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  23   1.833333   2.066667  0.883487   
72   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9   0   2.600000   2.300000  0.996164   
7

In [29]:
len(w)

199

In [30]:
# FP

len(w[w['균형'] == '균형'])

111

In [31]:
# TN

len(w[w['균형'] == '불균형'])

88

- 불균형인데 균형이라고 판단한 수는 111
- 불균형인데 불균형이라고 판단한 수는 88

# confusion matrix

0 |균형|불균형
-|-|-
균형|TP|FP
불균형|FN|TN


1. 정확도 accuracy = 실제(내) 데이터와 예측(기존방법)데이터가 얼마나 같은지를 평가하는 지표 = TP + TN / TP+TN+FP+FN
2. 정밀도 precision = 균형으로 예측한 데이터 중 실제로 균형인 데이터 비율 = TP / TP+FP
3. 재현율 recall = 실제 균형 중 균형으로 예측한 비율 = TP / TP+FN

=> 정밀도에 초점. (불균형을 균형으로 예측하여 재배치하지 않으면 서비스에 차질)


In [32]:
# 정확도

(len(z[z['균형'] == '균형']) + len(w[w['균형'] == '불균형'])) / len(df_new)

0.4236111111111111

In [33]:
# 정밀도

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(w[w['균형'] == '균형']))

0.23448275862068965

In [34]:
# 재현율

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(z[z['균형'] == '불균형']))

0.38202247191011235